### PR TITLE
[MIG] 15.0 base_technical_features

### PR DESCRIPTION
**Pull request by JuMiSanAr**
Migrate base_technical_features to 15.0

### CODE REVIEW

**Comment by simahawk** [comment]
@JuMiSanAr pls check the build

File: base_technical_features/readme/USAGE.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+.. figure:: static/description/user_preferences.png
+   :alt: User preferences
+
+Upon installation of this module, this preference is already
+set for the administrator user of the database.
+
+In the background, this preference is mapped to the *Technical feature (w/o
+debug mode)* group that this module adds. As an administrator, you can
+therefore manage this preference from the regular Users and Groups menu items.
+f it's not unique.
```
Comment (gurneyalex) [inline base_technical_features/readme/USAGE.rst]:
```suggestion
```

**Review by gurneyalex — APPROVED** [review]

File: .copier-answers.yml [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # Do NOT update manually; changes here will be overwritten by Copier
-_commit: v1.3.2
```
Comment (simahawk) [inline .copier-answers.yml]: have you ran `copier` to get this update?

**Comment by simahawk** [comment]
pre-commit is still :red_circle: 
I think it's better if we run a sync from the OCA template 1st, then rebase here.

**Comment by simahawk** [comment]
@JuMiSanAr I think you should drop your change on pre-commit conf and rebase on 15.0

File: .copier-answers.yml (hunk 2 above)
Comment (JuMiSanAr) [reply .copier-answers.yml]: Rebased on 15.0 instead.

**Review by simahawk — APPROVED** [review]

**Review by CasVissers-360ERP — APPROVED** [review]
Functional test.

**Comment by CasVissers-360ERP** [comment]
@JuMiSanAr 
I get an error when installing new modules after installing:

```
  File "/Users/360erp/PycharmProjects/customer/community/base_technical_features/models/res_users.py", line 54, in __init__
    type(self).SELF_READABLE_FIELDS += [
Exception

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/360erp/PycharmProjects/customer/odoo-15.0/odoo/http.py", line 643, in _handle_exception
    return super(JsonRequest, self)._handle_exception(exception)
  File "/Users/360erp/PycharmProjects/customer/odoo-15.0/odoo/http.py", line 301, in _handle_exception
    raise exception.with_traceback(None) from new_cause
TypeError: unsupported operand type(s) for +=: 'property' and 'list
```

**Comment by simahawk** [comment]
merge build failed w/ 
```
Error: reserveCache failed: Cache already exists. Scope: refs/heads/15.0-ocabot-merge-pr-391-by-simahawk-bump-nobump, Key: pre-commit-2-e61b05bccdb35a39be54b14e484996ab8f741257e3664b23cd4e167bcf9277cc-fbf4a408bf120b50899bf0e59355141f0e1e3a98ebc748ccf9df3a735f209967, Version: 28cdb9f5496f334116f23e86f0063f5d3a9348c2e22425a33171e071aadada7e
``` 
I'll try again later.
@JuMiSanAr can you check https://github.com/OCA/server-ux/pull/391#issuecomment-948543334 ?

File: base_technical_features/models/res_users.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+                    _("The user does not have access to technical " "features.")
+                )
+        if user.technical_features:
+            self.sudo().write({"groups_id": [(4, group.id)]})
+        else:
+            self.sudo().write({"groups_id": [(3, group.id)]})
+
+    def __init__(self, pool, cr):
+        super().__init__(pool, cr)
+        type(self).SELF_READABLE_FIELDS += [
```
Comment (BernatPForgeFlow) [inline base_technical_features/models/res_users.py]: TypeError: unsupported operand type(s) for +=: 'property' and 'list'

**Review by BernatPForgeFlow — CHANGES_REQUESTED** [review]

File: base_technical_features/models/res_users.py (hunk 3 above)
Comment (JuMiSanAr) [reply base_technical_features/models/res_users.py]: Updated to fix the bug.

**Comment by simahawk** [comment]
@BernatPForgeFlow can you check again pls?

**Review by CasVissers-360ERP — APPROVED** [review]
All good now 👍

**Pull request merged**